### DISCUSSION TITLE
How Can I Use IHttpClientFactory or IOptions<T> or Others in Stylet DI

### DISCUSSION THREAD

**Discussion by futugyou** [canton7/Stylet#320] [Q&A] [answered]
I want use  IHttpClientFactory  or others.
What should I do when I use Stylet Bootstrapper

**Comment by canton7** [comment, marked as answer]
You can either register them with Stylet's IoC container, or you can configure Stylet to use Microsoft.Extensions.DependencyInjection.

Which bit specifically are you having trouble with?

**Reply by futugyou** [reply]
my  app.xaml like this
```
<s:ApplicationLoader>
 <s:ApplicationLoader.Bootstrapper>
   <local:Bootstrapper/>
 </s:ApplicationLoader.Bootstrapper>
</s:ApplicationLoader>
```

and Bootstrapper is
```
public class Bootstrapper : MicrosoftDependencyInjectionBootstrapper<ShellViewModel>
```

when i run the code ,I got an error : “No service for type 'ShellViewModel' has been registered.”

**Reply by canton7** [reply]
Yeah, Microsoft.Extensions.DependencyInjection is a bit limited, as it can't auto-register types. You'll have to manually register each ViewModel in your bootstrapper's `ConfigureIoC` method

**Reply by futugyou** [reply]
I use IContainer in ShellViewModel,
but Container is an  internal class, how can I inject it? 

should i use ServiceProvider instead of IContainer ?

**Reply by canton7** [reply]
I'm not sure I follow. Do you mean you're using Stylet's `IContainer` in one of your ViewModels? If you've switched from using Stylet's IoC container to use Microsoft's IoC container, then you shouldn't use Stylet's IoC container, and that includes its `IContainer` type
